src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java

```diff
@@ -109,6 +116,8 @@ public SamlSecurityRealm(String signOnUrl, String idpMetadata, String displayNam
     if (usernameCaseConversion != null && !usernameCaseConversion.isEmpty()) {
       this.usernameCaseConversion = Util.fixEmptyAndTrim(usernameCaseConversion);
     }
+
+    this.userIdStrategy = userIdStrategy == null ? IdStrategy.CASE_INSENSITIVE : userIdStrategy;
```

Comment:
> is the forcing of case required?.

AFAIK First time you update the plugin with this change, in case you don't save anything on Configure Global security, then userIdStrategy will be null. Correct? Perhaps I am wrong...?